pkg/datastore/test/watch.go

```diff
@@ -829,10 +829,7 @@ func WatchEmissionStrategyTest(t *testing.T, tester DatastoreTester) {
 	features, err := ds.Features(ctx)
 	require.NoError(err)
 
-	expectsWatchError := false
-	if !(features.WatchEmitsImmediately.Status == datastore.FeatureSupported) {
-		expectsWatchError = true
-	}
+	expectsWatchError := (features.WatchEmitsImmediately.Status == datastore.FeatureSupported)
```

Comment:
Check this too

Reply:
This should be `expectsWatchError := (features.WatchEmitsImmediately.Status != datastore.FeatureSupported)`

Reply:
good catch!